### ISSUE TITLE
Wrong SparkSession initialization in Testing method 'withQbeastContextSparkAndTmpDir' 

### ISSUE THREAD

**Issue by eavilaes** [Qbeast-io/qbeast-spark#63]
**What went wrong?**
When running a **single test** that uses `withQbeastContextSparkAndTmpDir`, as there is no active `SparkSession` it fails. For example, in this test:

https://github.com/Qbeast-io/qbeast-spark/blob/35c9f5698ee7ce01e1eaf467013805dfad63d7da/src/test/scala/io/qbeast/spark/utils/QbeastDataSourceIntegrationTest.scala#L13-L15

The method `withQbeastContextSparkAndTmpDir` is called and will fail, due to `QbeastContext` needing an active `SparkSession`, which is not active at this point yet:
https://github.com/Qbeast-io/qbeast-spark/blob/35c9f5698ee7ce01e1eaf467013805dfad63d7da/src/test/scala/io/qbeast/spark/QbeastIntegrationTestSpec.scala#L144-L145

As you can see, `withQbeastContext` tries to get the SparkSession:
https://github.com/Qbeast-io/qbeast-spark/blob/35c9f5698ee7ce01e1eaf467013805dfad63d7da/src/test/scala/io/qbeast/spark/QbeastIntegrationTestSpec.scala#L126-L128

**How to reproduce?**
1. Code that triggered the bug, or steps to reproduce:
Run the mentioned test, for example, the one mentioned from `QbeastDataSourceIntegrationTest.scala`.

2. Branch and commit id:
main, 35c9f56

3. Spark version:
N/A

4. Hadoop version:
N/A

5. Are you running Spark inside a container? Are you launching the app on a remote K8s cluster? Or are you just running the tests in a local computer?
I'm running the tests in my local computer.

6. Stack trace:
```scala
No active or default Spark session found
java.lang.IllegalStateException: No active or default Spark session found
	at org.apache.spark.sql.SparkSession$.$anonfun$active$2(SparkSession.scala:1065)
	at scala.Option.getOrElse(Option.scala:189)
	at org.apache.spark.sql.SparkSession$.$anonfun$active$1(SparkSession.scala:1065)
	at scala.Option.getOrElse(Option.scala:189)
	at org.apache.spark.sql.SparkSession$.active(SparkSession.scala:1064)
	at io.qbeast.spark.QbeastIntegrationTestSpec.withQbeastContext$default$2(QbeastIntegrationTestSpec.scala:128)
	at io.qbeast.spark.QbeastIntegrationTestSpec.withQbeastContext$default$2$(QbeastIntegrationTestSpec.scala:128)
	at io.qbeast.spark.utils.QbeastDataSourceIntegrationTest.withQbeastContext$default$2(QbeastDataSourceIntegrationTest.scala:11)
	at io.qbeast.spark.QbeastIntegrationTestSpec.withQbeastContextSparkAndTmpDir(QbeastIntegrationTestSpec.scala:145)
	at io.qbeast.spark.QbeastIntegrationTestSpec.withQbeastContextSparkAndTmpDir$(QbeastIntegrationTestSpec.scala:144)
	at io.qbeast.spark.utils.QbeastDataSourceIntegrationTest.withQbeastContextSparkAndTmpDir(QbeastDataSourceIntegrationTest.scala:11)
	at io.qbeast.spark.utils.QbeastDataSourceIntegrationTest.$anonfun$new$1(QbeastDataSourceIntegrationTest.scala:15)
	at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
...
```

**Comment by osopardo1** [comment]
I suggest just going `withSparkAndTmpDir`, without involving any `QbeastContext` initialization in the spec. 

In case we are trying new configurations, we can do so with the `SparkSession.config(...) `

**Comment by eavilaes** [comment]
> I suggest just going `withSparkAndTmpDir`, without involving any `QbeastContext` initialization in the spec.
> 
> In case we are trying new configurations, we can do so with the `SparkSession.config(...) `

Then, when do we need `withQbeastContext()` in the tests? As I understand from your comment, we only need it if we change some of the `SparkConf`, but we can do it with the `SparkSession.config(...)`, right?

I quickly fixed it in the branch I was working (PR #62, commit 81f7b4) by changing the order of the methods, but I don't really know when we need the QbeastContext.

**Comment by osopardo1** [comment]
Closing because is already fixed in `main` since commit c4a6a64a6acd3ee18b3d38ba30e5609fd7f5b7b7
